### DISCUSSION TITLE
Codeium chat doesn't work

### DISCUSSION THREAD

**Discussion by Readf0x** [LazyVim/LazyVim#4144] [Q&A] [answered]
Codeium has chat disabled by default (though it does appear to function for others using a non lazy setup). The fact that it is not enabled in lazy does not surprise me, so I attempted to try and enable it. Unfortunately, I am experienced with neither lua or neovim, so I'm working blind here.
```lua
{
  "Exafunction/codeium.nvim",
  dependencies = {
    "nvim-lua/plenary.nvim",
  },
  cmd = "Codeium",
  build = ":Codeium Auth",
  opts = {},
  config = function()
    require("codeium").setup({
      enable_chat = true,
    })
  end,
},
{
  "nvim-cmp",
  dependencies = {
    {
      "Exafunction/codeium.nvim",
      cmd = "Codeium",
      build = "Codeium Auth",
      opts = {},
    },
  },
  opts = function(_, opts)
    table.insert(opts.sources, 1, {
      name = "codeium",
      group_index = 1,
      priority = 100,
    })
  end,
},
{
  "nvim-lualine/lualine.nvim",
  opts = function(_, opts)
    table.insert(opts.sections.lualine_x, 2, LazyVim.lualine.cmp_source("codeium"))
  end,
},
```
This is what I have set up, essentially just tacking the config function onto the default lazyvim config. However running `:Codeium Chat` results in the following error:
```
Error executing Lua callback: .../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:48: Vim:Error executing Lua callback: .../.local/share/nvim/lazy/codeium.nvim/lua/codeium/api.lua:168: attempt to concatenate upvalue 'port' (a nil value)
stack traceback:
	.../.local/share/nvim/lazy/codeium.nvim/lua/codeium/api.lua:168: in function 'request'
	.../.local/share/nvim/lazy/codeium.nvim/lua/codeium/api.lua:443: in function 'get_chat_ports'
	....local/share/nvim/lazy/codeium.nvim/lua/codeium/init.lua:25: in function <....local/share/nvim/lazy/codeium.nvim/lua/codeium/init.lua:19>
	[C]: in function 'cmd'
	.../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:48: in function <.../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:16>
stack traceback:
	[C]: in function 'cmd'
	.../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:48: in function <.../share/nvim/lazy/lazy.nvim/lua/lazy/core/handler/cmd.lua:16>
```
What am I doing wrong here? Where does the issue lie? Is it caused by mixing Lazyvim and Codeium, or is it a configuration error?
<hr>
Upon trying to use my setup in its current state, I've found that codeium isn't even functioning at the moment. Which is certainly just great. For the time being I'll have to go back to the builtin setup, but if anyone has any recommendations, that would be helpful.

**Comment by dpetka2001** [comment, marked as answer]
`Codeium` will only load if you first enter one time `insert` mode. After that you should be able to do `Codeium Chat` and launch the chat in your web browser.
